File: src/main/java/kurenai/mybot/CacheHolder.java
package kurenai.mybot;

import kurenai.mybot.cache.Cache;
import net.mamoe.mirai.message.data.OnlineMessageSource;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;

import java.util.concurrent.TimeUnit;

@Component
public class CacheHolder {

    public static final Cache<Integer, OnlineMessageSource> QQ_MSG_CACHE       = new Cache<>("QQ_MSG_CACHE");
    public static final Cache<Integer, Message>             TG_MSG_CACHE       = new Cache<>("TG_MSG_CACHE");
    public static final Cache<Integer, Integer>             TG_QQ_MSG_ID_CACHE = new Cache<>("TG_QQ_MSG_ID_CACHE");   // tg - qq message id cache;
    public static final Cache<Integer, Integer>             QQ_TG_MSG_ID_CACHE = new Cache<>("QQ_TG_MSG_ID_CACHE");   // qq - tg message id cache;

    public static void cache(OnlineMessageSource source, Message message) {
        var qqMsgId = source.getIds()[0];
        var tgMsgId = message.getMessageId();
        QQ_TG_MSG_ID_CACHE.put(qqMsgId, tgMsgId, 1, TimeUnit.DAYS);
        TG_QQ_MSG_ID_CACHE.put(tgMsgId, qqMsgId, 1, TimeUnit.DAYS);
        QQ_MSG_CACHE.put(qqMsgId, source, 1, TimeUnit.DAYS);
        TG_MSG_CACHE.put(tgMsgId, message, 1, TimeUnit.DAYS);
    }

}

File: src/main/kotlin/kurenai/mybot/CacheHolder.kt
package kurenai.mybot

import kurenai.mybot.cache.Cache
import mu.KotlinLogging
import net.mamoe.mirai.message.data.OnlineMessageSource
import org.telegram.telegrambots.meta.api.objects.Message
import java.util.concurrent.TimeUnit

object CacheHolder {

    private val log = KotlinLogging.logger {}

    val QQ_MSG_CACHE = Cache<Int?, OnlineMessageSource?>("QQ_MSG", 10000)
    val TG_MSG_CACHE = Cache<Int?, Message?>("TG_MSG", 10000)
    val TG_QQ_MSG_ID_CACHE = Cache<Int?, Int?>("TG_QQ_MSG_ID", 10000) // tg - qq message id cache;
    val QQ_TG_MSG_ID_CACHE = Cache<Int?, Int?>("QQ_TG_MSG_ID", 10000) // qq - tg message id cache;

    fun cache(source: OnlineMessageSource, message: Message) {
        val qqMsgId = source.ids[0]
        val tgMsgId = message.messageId

        QQ_TG_MSG_ID_CACHE.put(qqMsgId, tgMsgId, 7, TimeUnit.DAYS)
        TG_QQ_MSG_ID_CACHE.put(tgMsgId, qqMsgId, 7, TimeUnit.DAYS)
        QQ_MSG_CACHE.put(qqMsgId, source, 7, TimeUnit.DAYS)
        TG_MSG_CACHE.put(tgMsgId, message, 7, TimeUnit.DAYS)
    }
}